kaldb/src/test/java/com/slack/kaldb/metadata/service/ServicePartitionMetadataTest.java

```diff
@@ -79,4 +122,113 @@ public void testValidServicePartitionMetadata() {
         .isThrownBy(
             () -> new ServicePartitionMetadata(start.toEpochMilli(), end.toEpochMilli(), null));
   }
+
+  @Test
+  public void testOneServiceOnePartition() {
+    final String name = "testService";
+    final String owner = "serviceOwner";
+    final long throughputBytes = 1000;
+    final ServicePartitionMetadata partition = new ServicePartitionMetadata(100, 200, List.of("1"));
+
+    ServiceMetadata serviceMetadata =
+        new ServiceMetadata(name, owner, throughputBytes, List.of(partition));
+
+    serviceMetadataStore.createSync(serviceMetadata);
+    await().until(() -> serviceMetadataStore.listSync().size() == 1);
```

Comment:
In this case you probably want to change this to wait for `serviceMetadataStore.getCached()`. The `findPartitionsToQuery` call uses the service metadata store cache, but this is refreshed async. It's possible that you can have a `createSync` (or, just `create`), have it show up in a `listSync` call but not have it available in the cache.